sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BeamRowToStorageApiProto.java

```diff
@@ -272,6 +286,8 @@ private static Object messageValueFromRowValue(
     if (value == null) {
       if (fieldDescriptor.isOptional()) {
         return null;
+      } else if (fieldDescriptor.isRepeated()) {
```

Comment:
Currently we distinguish between the empty list and a missing value. I think we want to keep that distinction.

Reply:
Keeping the return value as null. 

It seems that the previous code will fail in the case of having a schema with a simple array of strings as a field and marking it as nullable. By adding this check here it also fixes that case. Adding a test to check those particularities.